Update 'samples/sample_extension' to Dart 2

chalin opened the pull request:
- Ran `dartfmt --fix`
- Fixed Dart 2 type errors
- Other misc cleanup (to conform to Effective Dart)

Contributes to https://github.com/dart-lang/site-www/issues/980

cc @a-siva @kwalrath @kevmoo

chalin commented on samples/sample_extension/test_sample_synchronous_extension.dart:
```diff
@@ -7,45 +7,44 @@ library test_sample_extension;
-    throw new StateError(message);
-  }
+  if (!condition) throw StateError(message);
+  print('$message - passed');
```
I added a print statement just because it felt reassuring to get _some_ output :)

chalin commented on samples/sample_extension/test/sample_extension_test_helper.dart:
```diff
@@ -6,9 +6,8 @@
-import 'dart:isolate';
 
-import "package:expect/expect.dart";
+import "package:test/test.dart";
```
I couldn't find an `expect` package in pub. Could `test` be used here?

chalin replied:
> expect is SDK-only

Ok, I'll revert that.

kevmoo commented:
Yes! expect is SDK-only

On Fri, Jun 29, 2018 at 9:59 AM Patrice Chalin <notifications@github.com>
wrote:

> *@chalin* commented on this pull request.
> ------------------------------
>
> In samples/sample_extension/test/sample_extension_test_helper.dart
> <https://github.com/dart-lang/sdk/pull/33711#discussion_r199221690>:
>
> >
> -import "package:expect/expect.dart";
> +import "package:test/test.dart";
>
> I couldn't find an expect package in pub. Could test be used here?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/dart-lang/sdk/pull/33711#pullrequestreview-133303513>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABCihdd9YUVQ3tFa8M4W4P7knvNfeX7ks5uBlzZgaJpZM4U9VFL>
> .
>

chalin committed:
Update 'samples/sample_extension' to Dart 2

- Ran dartfmt --fix
- Fixed Dart 2 type errors
- Other misc cleanup (to conform to Effective Dart)

Contributes to https://github.com/dart-lang/site-www/issues/980

chalin commented:
@a-siva - Under SDK 2-dev.66, the analyzer reports no issues (even with no implicit casts, and no implicit dynamic). The following tests pass:

- `dart test_sample_asynchronous_extension.dart`
- `dart test_sample_synchronous_extension.dart`

kevmoo commented:
See https://dart-review.googlesource.com/c/sdk/+/63260

chalin commented:
Any update on this?

kevmoo commented:
Need to comment here - https://dart-review.googlesource.com/c/sdk/+/63260 – now the source of truth

mit-mit commented:
@chalin @kevmoo what is the status of this PR?

kwalrath commented:
ping...

chalin committed:
sample_extension/test/sample_extension_test doesn't fail under dartk anymore

sstrickl commented:
Just a note, sample_extension/test/sample_extension_test still fails under dartk (see [this log](https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket.appspot.com/8927539872085951120/+/steps/vm_tests_shard_5/0/stdout) for example), so putting that back into the RuntimeError state in the status file for now.

chalin commented:
HI Stevie! Thanks for the note. Maybe if it fails intermittently, it would be best to skip that test?

sstrickl commented:
Perhaps! For now, I've at least marked it as possibly doing Pass and RuntimeError, and maybe we can figure out if there's a way to get rid of that latter while still catching any other type of failures in the meantime :)

chalin commented:
SGTM
